src/main/nativeMessaging.main.ts

```diff
@@ -87,22 +87,24 @@ export class NativeMessagingMain {
                 this.createWindowsRegistry('HKCU\\SOFTWARE\\Google\\Chrome', 'HKCU\\SOFTWARE\\Google\\Chrome\\NativeMessagingHosts\\com.8bit.bitwarden', path.join(destination, 'chrome.json'));
                 break;
             case 'darwin':
-                if (existsSync(`${this.homedir()}/Library/Application\ Support/Mozilla/NativeMessagingHosts/`)) {
-                    this.writeManifest(`${this.homedir()}/Library/Application\ Support/Mozilla/NativeMessagingHosts/com.8bit.bitwarden.json`, firefoxJson);
-                } else {
-                    this.logService.warning(`Firefox not found skipping.`);
-                }
-
-                if (existsSync(`${this.homedir()}/Library/Application\ Support/Google/Chrome/NativeMessagingHosts`)) {
-                    this.writeManifest(`${this.homedir()}/Library/Application\ Support/Google/Chrome/NativeMessagingHosts/com.8bit.bitwarden.json`, chromeJson);
-                } else {
-                    this.logService.warning(`Chrome not found skipping.`);
-                }
-
-                if (existsSync(`${this.homedir()}/Library/Application\ Support/Microsoft\ Edge/NativeMessagingHosts`)) {
-                    this.writeManifest(`${this.homedir()}/Library/Application\ Support/Microsoft\ Edge/NativeMessagingHosts/com.8bit.bitwarden.json`, chromeJson);
-                } else {
-                    this.logService.warning(`Microsoft Edge not found skipping.`);
+                const nmhs = {
```

Comment:
Nice refactor, I believe we can take it a step further and extract it to a function or field to avoid duplicating it in the install and uninstall parts.

Reply:
Yes, it's a good idea. I missed that